scripts/fault_domains_expr.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python3
+
+"""
+Generates an Azure Resource Manager (ARM) expression that will evaluate at
+deployment time to the number of fault domains available in a given location.
+Also generates Go code referencing the expression.
```

Comment:
This is a script that a human runs that emits copy-and-pasteable Go code. There is no automation, but the worst that can happen is an Azure location is upgraded from 2 to 3 fault domains, and there is a time period before we re-run the script and PR the changes where we fail to take advantage of that.

This could (should?) be rewritten in Go and added as a pre-flight check to `aks-engine deploy|generate`. Then we could regenerate the ARM expression just-in-time, although we would want to have a short timeout and fallback logic to ensure that never failed a deployment.